etk/extractors/date_extractor.py

```diff
@@ -383,20 +393,23 @@ def _parse_date(self, date_info: dict) -> Extraction or None:
                 p = self._symbol_list[date_info['order']][i] if not user_defined_pattern else user_defined_pattern[i]
                 if p in units['D']:
                     miss_day = False
-                if p in units['M']:
+                elif p in units['M']:
```

Comment:
Instead of miss_day/month/week/year variables, maybe a dictionary 
```
is_missing = {'D':True, 'M':True, 'Y': True, 'W':True}
```

and then you can loop:

```
for u in is_missing.keys() :
    if p in units[u]:
        is_missing[u] = False
        break
```